build.gradle

```diff
@@ -520,7 +529,13 @@ dependencyAnalysis {
                 severity('fail')
             }
             onUnusedDependencies {
-                exclude("junit:junit", "org.junit.jupiter:junit-jupiter-api", "org.junit.jupiter:junit-jupiter-engine", "com.fasterxml.jackson.core:jackson-annotations")
+                exclude("junit:junit", "org.junit.jupiter:junit-jupiter-api",
+                        "org.junit.jupiter:junit-jupiter-engine", "com.fasterxml.jackson.core:jackson-annotations",
+                        "org.springframework.boot:spring-boot-starter-oauth2-resource-server",
+                        "org.springdoc:springdoc-openapi-starter-webmvc-ui",
```

Comment:
I think the spring exclusions can be removed. with just one change later they are not needed - so dependencies are correct. That one change relates to making the springdoc dependency 'runtimeOnly'.

This is a reminder there are some other cleanups that should be done with gradle/ junit, but I will look at that once your spring change is merged

I am also ok if you want to merge as-is - I don't notice anything that would cause problems/regression

I will look at the junit/jupiter/jackson in a separate issue after merging - that isn't specific to your spring change